### ISSUE TITLE
Grails 2.5.5 does not work with Java 9

### ISSUE THREAD

**Issue by vbaarda** [grails/grails-core#10321]
Using grails 2.5.5 with Java 8, all is well:
```
$ $JAVA_HOME/bin/java -version
java version "1.8.0_111"
Java(TM) SE Runtime Environment (build 1.8.0_111-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.111-b14, mixed mode)
$ grails -version
Grails version: 2.5.5
```

After switching to Java 9, basic commands fail:
```
$ export JAVA_HOME=/usr/lib/jvm/jdk-9
$ $JAVA_HOME/bin/java -version
java version "9-ea"
Java(TM) SE Runtime Environment (build 9-ea+144)
Java HotSpot(TM) 64-Bit Server VM (build 9-ea+144, mixed mode)
$ grails -version
| Error An error occurred loading the grails-app/conf/BuildConfig.groovy file: Unable to make member of class sun.net.www.protocol.jar.JarURLConnection$JarURLInputStream accessible:  module java.base does not export sun.net.www.protocol.jar to unnamed module @71c3b41
$ grails create-app my-project
| Error An error occurred loading the grails-app/conf/BuildConfig.groovy file: Unable to make member of class sun.net.www.protocol.jar.JarURLConnection$JarURLInputStream accessible:  module java.base does not export sun.net.www.protocol.jar to unnamed module @71c3b41
```
### Steps to Reproduce

See above.

### Expected Behaviour

Grails commands should work.

### Actual Behaviour

Grails commands fail.

### Environment Information

- **Operating System**: Linux Mint 17 Qiana 64-bit
- **Grails Version:** 2.5.5
- **JDK Version:** build 9-ea+144 from Oracle
- **Container Version (If Applicable):** None

**Comment by vbaarda** [comment]
I downloaded the JDK from https://jdk9.java.net/download/

**Comment by graemerocher** [comment]
Stay with Java 8 or upgrade to Grails 3

**Comment by vbaarda** [comment]
@graemerocher 

Thanks -- good to know. Is that posted somewhere? All these searches turn up nothing:

https://groups.google.com/forum/#!searchin/grails-dev-discuss/%22java$209%22%7Csort:relevance
https://groups.google.com/forum/#!searchin/grails-dev-discuss/%22jdk$209%22%7Csort:relevance
https://www.google.com/search?q=%22java+9%22+site%3Agithub.com%2Fgrails
https://www.google.com/search?q=%22java+9%22+site%3Agrails.org
https://www.google.com/search?q=%22jdk+9%22+site%3Agithub.com%2Fgrails
https://www.google.com/search?q=%22jdk+9%22+site%3Agrails.org

**Comment by jeffbrown** [comment]
> Is that posted somewhere?

Probably not.  The docs at http://docs.grails.org/2.5.5/guide/gettingStarted.html#requirements say that Grails 2.5.5 requires 1.6 or above.  At the time that was written, that was true.  That could be updated to put an upper bound on the versions now that a version has been released that is not compatible with Grails 2.5.5.
